File: src/PAR.Domain/Entities/Group.cs
﻿using PAR.Domain.Common;

namespace PAR.Domain.Entities;

public class Group : BaseEntity
{
    public string Name { get; set; } = null!;
    public int TeacherId { get; set; }
    public int SchoolYearId { get; set; }
    public virtual SchoolYear? SchoolYear { get; set; }
    public virtual ApplicationUser? Teacher { get; set; }
    public virtual ICollection<Preschooler> Preschoolers { get; set; } = new HashSet<Preschooler>();
}

File: src/par-app/src/app/models/group.ts
import { BaseDto } from "./common/baseDto";
import {SchoolYearDto} from "./schoolYear";
import {ApplicationUser} from "./applicationUser";
import {PreschoolerDto} from "./preschooler";

export interface GroupDto extends BaseDto {
    name: string;
    teacherId: number;
    schoolYearId: number;
    schoolYear: SchoolYearDto | null;
    teacher: ApplicationUser | null;
    preschoolers: PreschoolerDto[];
}

export class GroupFormValues {
    id: number = 0;
    name: string = '';
    teacherId: number = 0;
    schoolYearId: number = 0;

    constructor(schoolYear?: GroupDto) {
        if (schoolYear) {
            this.id = schoolYear.id;
            this.name = schoolYear.name;
            this.teacherId = schoolYear.teacherId;
            this.schoolYearId = schoolYear.schoolYearId;
        }
    }
}